### DISCUSSION TITLE
After using the yalmip optimizer, the binary variables show 'NaN' but the ultimate result is normal.

### DISCUSSION THREAD

**Discussion by arielj-weibo** [yalmip/YALMIP#1108] [General discussion]
Comparing with the result of the model not using optimizer, the objective value of the model using optimizer is the same.Other variables are all the same.
However, the binary variables show 'NaN' while some other results are the same. I used the solver cplex in matlab.
I cannot figure out what's wrong with it.

%the variables
a=binvar(50,1);
b=binvar(50,1);
c=sdpvar(50,1);
%the parameter which needs to be parameterized
sdpvar d ;

constraints

objective value

optimizer

p(d)

double(a)
double(b)
%the above 2 shows NaN

**Comment by johanlofberg** [comment]
There is no official command called double, it is called value since a decade back.

However, value (or double...)  cannot be used when using optimizer. Solutions are returned by the object call.

**Reply by arielj-weibo** [reply]
Thank you for your reply.
I tried with value(a) value(b). The binary variables are still NaN.

I bother to ask you how to show the result of the binary variable. I don't understand _"Solutions are returned by the object call."_

**Reply by johanlofberg** [reply]
When you create the optimizer object, you define the solutions that should be returned. Those are the only solutions you get. value cannot be used

**Reply by arielj-weibo** [reply]
Thank you so much! I got it!
